dkg/dkg.go

```diff
@@ -48,6 +48,55 @@ type Config struct {
 	TestSigning bool
 }
 
+// exchanger is responsible for exchanging partial signatures between peers on libp2p.
+type exchanger struct {
+	setChan chan core.ParSignedDataSet
+	sigex   *parsigex.ParSigEx
+}
+
+func newExchanger(tcpNode host.Host, peerIdx int, peers []peer.ID) exchanger {
+	ex := exchanger{
+		sigex:   parsigex.NewParSigEx(tcpNode, peerIdx, peers),
+		setChan: make(chan core.ParSignedDataSet, len(peers)),
+	}
+
+	// Wiring core workflow components
+	ex.sigex.Subscribe(ex.getSets)
+
+	return ex
+}
+
+func (e *exchanger) exchange(ctx context.Context, set core.ParSignedDataSet) ([]core.ParSignedDataSet, error) {
+	err := e.sigex.Broadcast(ctx, core.Duty{}, set)
+	if err != nil {
+		return nil, err
+	}
+
+	var sets []core.ParSignedDataSet
+	sets = append(sets, set)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, err
+		case peerSet := <-e.setChan:
```

Comment:
how are you going to distinguish between lockhash and depositdata signatures...? (core workflow uses dutyType to distinguish, maybe do the same, by adding local duty types here `type dutyLock core.DutyType = 101`)

Reply:
and you probably need to do some fanout thing, where you have a router that routes signatures to the right duty type receive channel. 

Remember, different peers can send signatures at different times, some before this node and others after this node, and some might send lockhashes when this node is listenting for depositdatas...

Reply:
timing in distributed system is hard